consensus/consensus.go

```diff
@@ -31,35 +80,81 @@ func (pc *FastHotStuffConsensus) HandleTimeoutMessage( /* TODO */ ) {
 	// TODO
 }
 
-func (fc *FastHotStuffConsensus) HandleBlockProposal( /* TODO */ ) {
+func (fc *FastHotStuffConsensus) ConstructVoteQC( /* TODO */ ) {
+	// TODO
+}
+
+func (fc *FastHotStuffConsensus) ConstructTimeoutQC( /* TODO */ ) {
 	// TODO
 }
 
 func (fc *FastHotStuffConsensus) Start() {
+	fc.lock.Lock()
+	if fc.status == consensusStatusRunning {
+		// Nothing to do here. The consensus instance is already running.
+		fc.lock.Unlock()
+		return
+	}
+
+	// Update the consensus status to mark it as running.
+	fc.status = consensusStatusRunning
+	fc.lock.Unlock()
```

Comment:
we don't need to hold the lock while we update the next block construction time or the next timeout time?

Reply:
Ah good catch. We still need the lock